litestar/controller/generic.py

```diff
@@ -0,0 +1,226 @@
+# ruff: noqa: UP007, UP006
+from __future__ import annotations
+
+from inspect import isawaitable
+from operator import attrgetter
+from typing import TYPE_CHECKING, Any, Generic, List, Sequence, TypeVar, cast, get_args
+from uuid import UUID
+
+from typing_extensions import get_origin, get_type_hints
+
+from litestar.connection.request import Request
+from litestar.controller.base import Controller
+from litestar.enums import HttpMethod
+from litestar.exceptions import ImproperlyConfiguredException
+from litestar.handlers import BaseRouteHandler, HTTPRouteHandler
+from litestar.repository.filters import CollectionFilter
+from litestar.types import Empty
+from litestar.utils import is_class_and_subclass
+from litestar.utils.typing import get_safe_generic_origin
+
+if TYPE_CHECKING:
+    from litestar.dto import AbstractDTO
+    from litestar.repository import AbstractAsyncRepository, AbstractSyncRepository
+    from litestar.types import EmptyType
+
+ModelT = TypeVar("ModelT")
+IdAttrT = TypeVar("IdAttrT", str, int, UUID)
+
+GENERIC_METHOD_NAMES = (
+    "create_instance",
+    "update_instance",
+    "delete_instance",
+    "get_instance",
+    "create_many",
+    "update_many",
+    "delete_many",
+    "get_many",
+)
+
+
+class GenericController(Controller, Generic[ModelT, IdAttrT]):
```

Comment:
I'd prefer an API were the controller has 1 generic type only. I can see why you opted for `IdAttrT`, would it be possible to extract the type of id attribute using `ModelT` and `id_attribute`.

`GenericController[MyModel]` looks much better that `GenericControlller[MyModel, str]`

Reply:
i went with your input, this is the API now